File: ProyectoFinal/Nomina/C++/Constructor.cpp
#include <iostream>
#include <cstdlib>

/**
*
*@author Laura Santacruz
*
**/

using namespace std;

class Nomina{
	
	//Atributos
	string nombres,apellidos;
    int cedula,salario;
    
	public:
		//Constructor
		Nomina(string nom,string ape, int ced, int sal)
		{
			this->nombres=nom;
        	this->apellidos=ape;
        	this->cedula=ced;
        	this->salario=sal;
		}
		void Datos()
		{
			cout<<"El nombre del empleado es:"<<nombres;
			cout<<endl;
        	cout<<"con el apellido: "<<apellidos;
        	cout<<endl;
        	cout<<"la cedula del empleado es: "<<cedula;
        	cout<<endl;
        	cout<<"y recibe un salario de : $ "<<salario;
        	cout<<endl;
		}		
};

class RecursosHumanos{
	
	string sex,estado;
    int hijo;
    
	public:
		RecursosHumanos(string genero,string estci, int hijos)
		{
			this->sex=genero;
        	this->estado=estci;
        	this->hijo= hijos;
		}	
	public:
		void recursos()
    	{
     		cout<<"El genero del empleado es : "<<sex;
     		cout<<endl;
     		cout<<"El estado civil de el empleado es:  "<<estado;
    		cout<<endl;
			cout<<"\nY tiene esta cantidad de hijos: "<<hijo;
    		cout<<endl;
		
		}	
};


int main(){
	
	//Datos de nomina
	Nomina empleado1("juanito", "Alimaña", 307803030, 57907870), empleado2("Daniel","Andrade",123458,80000000);
	Nomina empleado3("Martha","Ojeda",45810201,1000000), empleado4("Sergio","Cadena",564213156,2000000), empleado5("Laura","Santacruz",98091000,4512313);
	
	//Datos de Recursos humanos
	RecursosHumanos empleadoA("Masculino", "Soltero", 2), empleadoB("Masculino", "Casado", 3);
	RecursosHumanos empleadoC("Femenino", "Casada", 3), empleadoD("Masculino", "Casado", 5), empleadoE("Femenino", "Soltera", 0);
	
	empleado1.Datos();
	empleadoA.recursos();

	empleado2.Datos();
	empleadoB.recursos();
	
	empleado3.Datos();
	empleadoC.recursos();
	
	empleado4.Datos();
	empleadoD.recursos();
	
	empleado5.Datos();
	empleadoE.recursos();


}

File: ProyectoFinal/Nomina/Java/Constructor.java
/*
 *
 * Esta clase crea los objetos a invocar por parte de clase nomina. la cual 
 * lleva los atributos del empleado y su salario
 */
package Constructores;

/**
 *
 * @author Laura Santacruz
 */
public class Constructor {
    public static void main(String[] args) {
        //Creando objetos
        
       Nomina empleado1 = new Nomina("juanito", "Alimaña", 307803030, 57907870);
       empleado1.Datos();
       RecursosHumanos empleado = new RecursosHumanos ("Masculino", "Soltero", 2);
       empleado.recursos();
        
       Nomina empleado2 = new Nomina ("Daniel","Andrade",123458,80000000);
       empleado2.Datos();
       RecursosHumanos empleadoA = new RecursosHumanos ("Masculino", "Casado", 3);
       empleadoA.recursos();
        
       Nomina empleado3 = new Nomina ("Martha","Ojeda",45810201,1000000);
       empleado3.Datos();
       RecursosHumanos empleadoB = new RecursosHumanos ("Femenino", "Casada", 3);
       empleadoB.recursos();
         
       System.out.println("");
       
       Nomina empleado4 = new Nomina ("Sergio","Cadena",564213156,2000000);
       empleado4.Datos();
       RecursosHumanos empleadoC = new RecursosHumanos ("Masculino", "Casado", 5);
       empleadoC.recursos();
       
       Nomina empleado5 = new Nomina ("Laura","Santacruz",98091000,4512313);
       empleado5.Datos();
       RecursosHumanos empleadoD = new RecursosHumanos ("Femenino", "Soltera", 0);
       empleadoD.recursos();
    }  
}
